### PR TITLE
fix: add error body to 500 responses

### PR DESCRIPTION
**Pull request by notanatol**
### Checklist

- [X] I have read the [coding guide](https://github.com/ethersphere/bee/blob/master/CODING.md)
- [ ] My change requires a documentation update and I have done it

### Description
Closes: #2026

---
This change is [Reviewable](https://reviewable.io/reviews/ethersphere/bee/3148)

### CODE REVIEW

**Review by istae — APPROVED** [review]

File: pkg/api/bytes.go [hunk 1]
```diff
@@ -37,13 +37,13 @@ func (s *Service) bytesUploadHandler(w http.ResponseWriter, r *http.Request) {
 	if err != nil {
 		logger.Debugf("bytes upload: get putter:%v", err)
 		logger.Error("bytes upload: putter")
-		jsonhttp.BadRequest(w, nil)
+		jsonhttp.BadRequest(w, fmt.Sprintf("new stamper putter: %v", err))
```
Comment (aloknerurkar) [inline pkg/api/bytes.go:40]: The reason why this was not there in the first place was that we did not want to leak all these implementation details to the users. For eg, stamper putter or split write etc don't mean anything to the users. I am not sure if we want to expose these things.

Instead, we should put some messages here that don't leak the implementation details. This is the same reason why the error logs dont log the actual errors. As these errors are not meant for users. This is why they only show in debug logs.

Comment (notanatol) [reply pkg/api/bytes.go:40]: reverted

**Review by mrekucci — CHANGES_REQUESTED** [review]
Exposing internal errors directly to the consumer is bad practice. Errors are not sanitized and can leak sensitive information. Also, exposing internal errors to the consumer does not provide much value; instead, messages should be worded in such a way that they do not go into unnecessary detail and multiple 500 API errors returned by the same endpoint can be easily distinguished: https://cloud.google.com/apis/design/errors

> Reviewed 13 of 16 files at r1, 3 of 3 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3148#-:-N8XWGGf6y3eOy7jM-GC:b-dlibhu)* status: all files reviewed, 1 unresolved discussion (waiting on @aloknerurkar)

**Comment by notanatol** [comment]
> Exposing internal errors directly to the consumer is bad practice. Errors are not sanitized and can leak sensitive information. Also, exposing internal errors to the consumer does not provide much value; instead, messages should be worded in such a way that they do not go into unnecessary detail and multiple 500 API errors returned by the same endpoint can be easily distinguished: https://cloud.google.com/apis/design/errors
> 
> > Reviewed 13 of 16 files at r1, 3 of 3 files at r2, all commit messages.
> > _[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3148#-:-N8XWGGf6y3eOy7jM-GC:b-dlibhu)_ status: all files reviewed, 1 unresolved discussion (waiting on @aloknerurkar)

that's a valid comment but I believe this discussion should be had here: https://github.com/ethersphere/bee/issues/2026

**Review by mrekucci — APPROVED** [review]
> Reviewed 16 of 16 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/ethersphere/bee/3148)* status: all files reviewed, 1 unresolved discussion (waiting on @aloknerurkar)

**Pull request merged**